kolibri/core/tasks/job.py

```diff
@@ -249,10 +249,18 @@ def save_meta(self):
         self.storage.save_job_meta(self)
 
     def update_progress(self, progress, total_progress):
-        if self.track_progress:
-            self.progress = progress
-            self.total_progress = total_progress
-            self.storage.update_job_progress(self.job_id, progress, total_progress)
+        if not self.track_progress:
+            return
+        if not isinstance(progress, (int, float)) or not isinstance(
+            total_progress, (int, float)
+        ):
+            logger.warning(
+                f"Tried to set invalid progress values on job {self.job_id} for task {self.func} with progress: {progress} and total_progress {total_progress}"
+            )
```

Comment:
Would it be helpful here to raise a ValueError here? I can see myself missing the warning in the logs if I were coming at this while debugging in progress updating on the front-end or something. Not a blocking comment - just a thought.

Reply:
My reason for only logging is that given we have seen this in the wild (as per the issue) I don't want to cause existing jobs to fail just because of a progress tracking error.